### ISSUE TITLE
verifytypes doesn't work with packages that are installed as editable

### ISSUE THREAD

**Issue by tkukushkin** [microsoft/pyright#5473]
**Describe the bug**
I tried to use pyright --verifytypes with [pystatsd](https://github.com/jsocol/pystatsd). 

When I install it **not** as editable (`pip install .`), pyright finds statsd module properly:

```
$ pyright --verifytypes statsd
Module name: "statsd"
Package directory: "/Users/tkukushkin/Projects/pystatsd/.venv/lib/python3.10/site-packages/statsd"
Module directory: "/Users/tkukushkin/Projects/pystatsd/.venv/lib/python3.10/site-packages/statsd"
Path of py.typed file: "/Users/tkukushkin/Projects/pystatsd/.venv/lib/python3.10/site-packages/statsd/py.typed"
...
```

But when I install the same package as editable (`pip install -e .`), pyright can't find the module:

```
$ pyright --verifytypes statsd
Module name: "statsd"
Package directory: ""
Module directory: ""
   error: Module "statsd" cannot be resolved
```

**To Reproduce**
```bash
gh repo clone jsocol/pystatsd
cd pystatsd
pip install -e .
pyright --verifytypes statsd
```

**Expected behavior**
Pyright can find the module in both cases.

**VS Code extension or command-line**
pyright CLI, version – 1.1.316

**Comment by erictraut** [comment]
Please refer to [this documentation](https://microsoft.github.io/pyright/#/import-resolution?id=editable-installs).

**Comment by tkukushkin** [comment]
Oh, thank you, it helped! I apologize for the inconvenience

**Comment by wookayin** [comment]
TL;DR from https://microsoft.github.io/pyright/#/import-resolution?id=editable-installs :

Install package (editable) with the ["legacy behavior"](https://setuptools.pypa.io/en/latest/userguide/development_mode.html#legacy-behavior)
```
pip install -e . --config-settings editable_mode=compat
# or, alternatively
pip install -e . --config-settings editable_mode=strict
```
